Relax dependency requirements for google test library and allow customization of cctz target

romange opened the pull request.

romange committed:
Relax dependency on googletest and allow customization of cctz target

romange committed:
Typo fix

romange commented:
The default behavior was not changed

ahedberg commented:
Hey @romange, thanks for the PR! Many of us are out for the holiday break, so you may not see much movement here until the new year. Assigning to @JonathanDCohen so this doesn't get lost in the meantime.

JonathanDCohen commented:
FInally getting around to checking this -- sorry for the long delay.  Could you, both for reviewer's and abseil users' sakes, update the CMake Readme file which shows the corresponding change to users code that goes with this CL?

romange committed:
Update README.md

romange commented:
Hi @JonathanDCohen, I updated README file and fixed a typo in CMakeLists.txt. Now it should fill correctly the default value into variable `ABSL_CCTZ_TARGET`.

JonathanDCohen commented on absl/time/CMakeLists.txt:
```diff
@@ -32,7 +32,7 @@ list(APPEND TIME_SRC
   ${TIME_INTERNAL_HEADERS}
 )
-set(TIME_PUBLIC_LIBRARIES absl::base absl::stacktrace absl::int128 cctz)
+set(TIME_PUBLIC_LIBRARIES absl::base absl::stacktrace absl::int128 {ABSL_CCTZ_TARGET})
```
"${ABSL_CCTZ_TARGET}"

JonathanDCohen commented:
Sorry if I'm being dense, but what does the extra ABSL_CCTZ_TARGET buy users?  It allows them to have  cctz installed under whatever name they choose?

Just making sure I'm on the same page as this change.

romange committed:
Fix a typo

romange commented:
Consider the case where users already have cctz library in their build
system but it's not under "cctz" target name.
For example, in my case, the build system declares all third-party library
targets with "TRDP::" prefix in their names.

On Mon, Jan 29, 2018 at 10:49 PM, Jonathan Cohen <notifications@github.com>
wrote:

> Sorry if I'm being dense, but what does the extra ABSL_CCTZ_TARGET buy
> users? It allows them to have cctz installed under whatever name they
> choose?
>
> Just making sure I'm on the same page as this change.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/abseil/abseil-cpp/pull/69#issuecomment-361381647>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADgSiPo6hrWaB4b_Psyau-jaVF89dZ_jks5tPi72gaJpZM4RLep0>
> .
>



-- 
Best regards,
     Roman

JonathanDCohen commented on CMakeLists.txt:
```diff
@@ -58,29 +58,34 @@ list(APPEND ABSL_COMMON_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR})
+  check_target(gmock)
+
+  # -fexceptions
+  set(ABSL_EXCEPTIONS_FLAG "${CMAKE_CXX_EXCEPTIONS}")
```
We need exceptions even in non-debug builds

JonathanDCohen commented:
That was what I figured, thanks for clarifying.

romange committed:
Move ABSL_EXCEPTIONS_FLAG define

romange commented:
Ooops, sorry for all these bugs. Fixed.

manshreck commented on CMake/README.md:
```diff
@@ -29,7 +29,8 @@
     the targets  `gtest`, `gtest_main`, `gmock` and `cctz` need
     to be declared in your project before including abseil with `add_subdirectory`.
-
+    However, if abseil is compiled with -DBUILD_TESTING=OFF, then `gtest`, `gmock` and `gtest_main` are not required.
```
Wrap at 80 characters and enclose flags in backticks:

`-DBUILD_TESTING=OFF`

manshreck commented on CMake/README.md:
```diff
@@ -29,7 +29,8 @@
     to be declared in your project before including abseil with `add_subdirectory`.
-
+    However, if abseil is compiled with -DBUILD_TESTING=OFF, then `gtest`, `gmock` and `gtest_main` are not required.
+    In addition, it's possible to override the name of cctz target to custom one with option -DABSL_CCTZ_TARGET=my_cctz
```
name of -> name of the `cctz` target to a custom one using the option `-DABSL_CCTZ_TARGET=*my_cctz*`

manshreck reviewed (changes requested):
Some nits on the docs. Please wrap at 80 columns.

romange committed:
Doc fixed

romange commented:
@manshreck  ptal.
